frontend/app/routes/$lang+/_public+/apply+/$id+/applicant-information.tsx

```diff
@@ -50,14 +55,40 @@ export async function loader({ request, params }: LoaderFunctionArgs) {
 export async function action({ request, params }: ActionFunctionArgs) {
   const applyFlow = getApplyFlow();
   const { id } = await applyFlow.loadState({ request, params });
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  /**
+   * Schema applicant information.
+   */
+  const applicantInformationSchema: z.ZodType<ApplicantInformationState> = z.object({
+    socialInsuranceNumber: z
+      .string()
+      .trim()
+      .refine(isValidSin, { message: t('apply:applicant-information.error-message.valid-sin') }),
+    firstName: z
+      .string()
+      .trim()
+      .min(1, { message: t('apply:applicant-information.error-message.first-name') }),
+    lastName: z
+      .string()
+      .trim()
+      .min(1, { message: t('apply:applicant-information.error-message.last-name') }),
+    maritalStatus: z.string({ required_error: t('apply:applicant-information.error-message.marital-status') }),
```

Comment:
Here's my suggestion since the error message should be the same with the following scenarios:

- If the field is not defined (`undefined`), it displays the `required_error` message.
- If the field type is not `string`, specifically triggered when `null` is returned due to no marital statuses being selected, it shows the `invalid_type_error` message.
- If the selected marital status value is empty, it presents the `min` message.

```suggestion
    maritalStatus: z
      .string({ errorMap: () => ({ message: t('apply:applicant-information.error-message.marital-status') }) })
      .trim()
      .min(1, t('apply:applicant-information.error-message.marital-status')),
```

ATM, here's the error message that appears when the user hasn't selected anything:

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/114004123/e1a27295-4e83-47db-a316-07db7257933d)

Reply:
To only leverage on `.min()`, I highly suggest your convert the returned value from `FormData.get()` to the corresponding type:

```tsx
const data = {
    socialInsuranceNumber: String(formData.get('socialInsuranceNumber') ?? ""),
    firstName: String(formData.get('firstName') ?? ""),
    lastName: String(formData.get('lastName') ?? ""),
    maritalStatus: String(formData.get('maritalStatus') ?? ""),
  };
```

**_Note:_** ` ?? ""` is added to fallback to an empty string when `null` is returned.

Reply:
okay, fair points.